## Introduction and background
  
  BellaBeat is a successful small company in the global smart device market.The Chief Creative Officer of Bellabeat want to analyze smart device fitness data in order to unlock new growth opportunities for the company.This analysis will focus on one of Bellabeat’s products and analyze smart device data to gain insight into how consumers are using their smart devices. The insights  will then help guide marketing strategy for the company.
  

 ## Installing and loading common packages

In [124]:
import pandas as pd 
import seaborn as sns
import matplotlib as plt

## Upload CSV files and creating Dataframe

In [125]:
dailyActivity_df = pd.read_csv('BellaBeat_Data/dailyActivity_merged.csv')
dailyActivity_df.info()
dailyActivity_df.head()
total_unique = dailyActivity_df['Id'].unique()
len(total_unique)



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

33

In [126]:
dailyCalories_df = pd.read_csv('BellaBeat_Data/dailyCalories_merged.csv')
dailyCalories_df.info()
dailyCalories_df.head()
total_unique = dailyCalories_df['Id'].unique()
len(total_unique)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Id           940 non-null    int64 
 1   ActivityDay  940 non-null    object
 2   Calories     940 non-null    int64 
dtypes: int64(2), object(1)
memory usage: 22.2+ KB


33

In [127]:
dailyIntensities_df = pd.read_csv('BellaBeat_Data/dailyIntensities_merged.csv')
dailyIntensities_df.info()
dailyIntensities_df.head()
total_unique = dailyIntensities_df['Id'].unique()
len(total_unique)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 10 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDay               940 non-null    object 
 2   SedentaryMinutes          940 non-null    int64  
 3   LightlyActiveMinutes      940 non-null    int64  
 4   FairlyActiveMinutes       940 non-null    int64  
 5   VeryActiveMinutes         940 non-null    int64  
 6   SedentaryActiveDistance   940 non-null    float64
 7   LightActiveDistance       940 non-null    float64
 8   ModeratelyActiveDistance  940 non-null    float64
 9   VeryActiveDistance        940 non-null    float64
dtypes: float64(4), int64(5), object(1)
memory usage: 73.6+ KB


33

In [128]:
dailySteps_df = pd.read_csv('BellaBeat_Data/dailySteps_merged.csv')
dailySteps_df.info()
dailySteps_df.head()
total_unique = dailySteps_df['Id'].unique()
len(total_unique)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Id           940 non-null    int64 
 1   ActivityDay  940 non-null    object
 2   StepTotal    940 non-null    int64 
dtypes: int64(2), object(1)
memory usage: 22.2+ KB


33

In [129]:
sleepDay_df = pd.read_csv('BellaBeat_Data/SleepDay_merged.csv')
sleepDay_df.info()
sleepDay_df.head()
total_unique = sleepDay_df['Id'].unique()
len(total_unique)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 413 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  413 non-null    int64 
 1   SleepDay            413 non-null    object
 2   TotalSleepRecords   413 non-null    int64 
 3   TotalMinutesAsleep  413 non-null    int64 
 4   TotalTimeInBed      413 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 16.3+ KB


24

In [130]:
weightLogInfo_df = pd.read_csv('BellaBeat_Data/weightLogInfo_merged.csv')
weightLogInfo_df.info()
weightLogInfo_df.head()
total_unique = weightLogInfo_df['Id'].unique()
len(total_unique)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              67 non-null     int64  
 1   Date            67 non-null     object 
 2   WeightKg        67 non-null     float64
 3   WeightPounds    67 non-null     float64
 4   Fat             2 non-null      float64
 5   BMI             67 non-null     float64
 6   IsManualReport  67 non-null     bool   
 7   LogId           67 non-null     int64  
dtypes: bool(1), float64(4), int64(2), object(1)
memory usage: 3.9+ KB


8

## Overview of our Data
overall all our data are organised in columns and rows and all have as common column the Id column. The number of participants for each data set used are 33 except for the sleep_Day and WeightLogInfo. the sleep_day has 24 participants which the results can still be relevant to our analysis but for the weightLogInfo with only 8 participants, the information collected will not be enough to make useful analysis.

## Exploring some Statistics

In [131]:
pd.set_option('display.float_format', lambda x:'%.2f'% x)
dailyActivity_df['TotalSteps'].describe()


count     940.00
mean     7637.91
std      5087.15
min         0.00
25%      3789.75
50%      7405.50
75%     10727.00
max     36019.00
Name: TotalSteps, dtype: float64

As per CDC recommendation taking 8,000 steps per day was associated with a 51% lower risk for all-cause mortality. The participant in this study are a little bit under this threshold with 7637 steps a day

In [132]:
pd.set_option('display.float_format', lambda x:'%.2f'% x)
dailyCalories_df['Calories'].describe()


count    940.00
mean    2303.61
std      718.17
min        0.00
25%     1828.50
50%     2134.00
75%     2793.25
max     4900.00
Name: Calories, dtype: float64

FDA recommendation for calorie intake for adult is usually 2000 calories, however this number can be higher or lower depending on several factors such as age, sex, weight, height... so with 2303 calories on average we can say the participants are fairly within the norm

In [133]:
pd.set_option('display.float_format', lambda x:'%.2f'% x)
dailyIntensities_df['SedentaryMinutes'].describe()


count    940.00
mean     991.21
std      301.27
min        0.00
25%      729.75
50%     1057.50
75%     1229.50
max     1440.00
Name: SedentaryMinutes, dtype: float64

With 991 minutes or roughly 16.5 hours of sedentary activity we can conclude the participants in this study are not very active

In [134]:
pd.set_option('display.float_format', lambda x:'%.2f'% x)
dailySteps_df['StepTotal'].describe()


count     940.00
mean     7637.91
std      5087.15
min         0.00
25%      3789.75
50%      7405.50
75%     10727.00
max     36019.00
Name: StepTotal, dtype: float64

In [135]:
pd.set_option('display.float_format', lambda x:'%.2f'% x)
sleepDay_df['TotalMinutesAsleep'].describe()


count   413.00
mean    419.47
std     118.34
min      58.00
25%     361.00
50%     433.00
75%     490.00
max     796.00
Name: TotalMinutesAsleep, dtype: float64

The participants seems to be getting just enough sleep to meet CDC recommendation of 7 hours sleep a day